specs/order_spec.rb

```diff
@@ -1,110 +1,293 @@
+      @normal_order.add_product("sandwich", 4.25)
 
-      order.add_product("sandwich", 4.25)
-      order.products.include?("sandwich").must_equal true
+      @normal_order.products.include?("sandwich").must_equal true # assertion
+    end
+
+    it "Returns true if the product is new" do
+      @normal_order.add_product("salad", 4.25).must_equal true # assertion
     end
 
     it "Returns false if the product is already present" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
+      before_total = @normal_order.total
+      result_of_add = @normal_order.add_product("banana", 4.25) # already has
+      after_total = @normal_order.total
 
-      order = Grocery::Order.new(1337, products)
-      before_total = order.total
+      result_of_add.must_equal false # assertion
+      before_total.must_equal after_total # assertion
+    end
+  end
 
-      result = order.add_product("banana", 4.25)
-      after_total = order.total
+  describe "#remove_product" do
+    # Tests if add_product increases products size.
+    it "Decreases the number of products" do
+      @normal_order.remove_product("banana")
+      expected_count = @normal_products.size - 1
 
-      result.must_equal false
-      before_total.must_equal after_total
+      @normal_order.products.count.must_equal expected_count # assertion
     end
 
-    it "Returns true if the product is new" do
-      products = { "banana" => 1.99, "cracker" => 3.00 }
-      order = Grocery::Order.new(1337, products)
+    # Tests if program adds new product to products.
+    it "Is added to the collection of products" do
+      @normal_order.remove_product("banana")
 
-      result = order.add_product("salad", 4.25)
-      result.must_equal true
+      @normal_order.products.include?("banana").must_equal false # assertion
+    end
+
+    it "Returns true if the product is removed" do
+      @normal_order.remove_product("banana").must_equal true # assertion
+    end
+
+    it "Returns false if the product isn't a product" do
+      before_total = @normal_order.total
+      result_of_remove = @normal_order.remove_product("corn") # doesn't have
+      after_total = @normal_order.total
+
+      result_of_remove.must_equal false # assertion
+      before_total.must_equal after_total # assertion
     end
   end
+
 end
 
-# TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+
+describe "Order Wave 2" do
+
+  expected_first_id = 1
+  expected_first_products =
+    {"Slivered Almonds"=>22.88, "Wholewheat flour"=>1.93, "Grape Seed Oil"=>74.9}
+
+  expected_last_id = 100
+  expected_last_products =
+    {"Allspice"=>64.74, "Bran"=>14.72, "UnbleachedFlour"=>80.59}
+
+
   describe "Order.all" do
+
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
+      all_orders = Grocery::Order.all
+
+      all_orders.must_be_kind_of Array # assertion
+      assert all_orders.all? { |order| order.class == Grocery::Order} # assertion
```

Comment:
Nice, but this is **assert-style** testing.  Instead of doing this, it would be better to:

```ruby
(all_orders.all? { |order| order.class == Grocery::Order}).must_equal true
```

Just to stay consistent.